kafka-client/src/main/java/dev/responsive/kafka/internal/stores/ResponsiveGlobalStore.java

```diff
@@ -87,60 +74,61 @@ public void init(final ProcessorContext context, final StateStore root) {
   }
 
   @Override
-  public void init(final StateStoreContext context, final StateStore root) {
+  public void init(final StateStoreContext storeContext, final StateStore root) {
     try {
       log.info("Initializing global state store {}", name);
-      this.context = (GlobalProcessorContextImpl) context;
+
+      final SharedClients sharedClients = loadSharedClients(storeContext.appConfigs());
+
       // this is bad, but the assumption is that global tables are small
-      // and can fit in a single partition - all writers will write using
-      // the same partition key. we should consider using ALLOW FILTERING
+      // and can fit in a single topicPartition - all writers will write using
+      // the same topicPartition key. we should consider using ALLOW FILTERING
       // instead. that would make gets/puts more efficient and the queries
-      // to be more evently distributed and take the hit only on range
-      // queries
-      partition = 0;
-      final SharedClients sharedClients = SharedClients.loadSharedClients(context.appConfigs());
-      client = sharedClients.cassandraClient;
-
-      schema = client.prepareKVTableSchema(params);
-      log.info("Global table {} is available for querying.", name);
+      // to be more evently distributed and take the hit only on range queries
+      final TopicPartition globalTopicPartition =
+          new TopicPartition(((GlobalProcessorContextImpl) storeContext).topic(), 0);
+
+      super.init(
+          storeContext,
+          sharedClients.cassandraClient,
+          InternalConfigs.loadStoreRegistry(storeContext.appConfigs()),
+          globalTopicPartition
+      );
 
-      schema.init(name.cassandraName(), SubPartitioner.NO_SUBPARTITIONS, partition);
+      schema = sharedClients.cassandraClient.prepareKVTableSchema(params);
 
-      open = true;
+      schema.init(
+          name.cassandraName(),
+          SubPartitioner.NO_SUBPARTITIONS,
+          changelog.partition()
+      );
+      log.info("Global table {} is available for querying.", name);
 
-      context.register(root, (key, value) -> {
-        if (value == null) {
-          delete(new Bytes(key));
-        } else {
-          put(new Bytes(key), value);
+      super.open(root, batch -> {
+        for (final ConsumerRecord<byte[], byte[]> record : batch) {
+          put(Bytes.wrap(record.key()), record.value());
         }
       });
+
     } catch (InterruptedException | TimeoutException e) {
       throw new ProcessorStateException("Failed to initialize store.", e);
     }
   }
 
-  @Override
-  public boolean isOpen() {
-    return open;
-  }
-
-  @Override
-  public boolean persistent() {
-    return false;
-  }
-
   @Override
   public void put(final Bytes key, final byte[] value) {
-    client.execute(schema.insert(name.cassandraName(), partition, key, value, context.timestamp()));
-    StoreQueryUtils.updatePosition(position, context);
+    if (value == null) {
```

Comment:
nit: thanks for the fix! let's do this in a separate PR so this one is just a refactor

Reply:
fair enough -- https://github.com/responsivedev/responsive-pub/pull/154